compiler/AST/build.cpp

```diff
@@ -1035,6 +1035,9 @@ buildForallLoopExpr(Expr* indices, Expr* iteratorExpr, Expr* expr, Expr* cond, b
 //
 Expr* buildForallLoopExprFromArrayType(CallExpr* buildArrTypeCall,
                                            bool recursiveCall) {
+  if (buildArrTypeCall->isPrimitive(PRIM_ERROR))  // ex. 'type T = [];'
+    return buildArrTypeCall;
+
```

Comment:
Might be good to put an INT_ASSERT that it's not another type of primitive, since I'm assuming that would also crash the compiler on line 1042

Reply:
Turns out Line 1042 is guarded against `baseExpr` being a primitive. So no change is needed there.

Reply:
How so?  Is the PRIM_ERROR case on line 1038 the only one that would be allowed into this function?

Reply:
Oops, I looked at the wrong line.

The line `UnresolvedSymExpr* ursym = toUnresolvedSymExpr(buildArrTypeCall->baseExpr);` will result in `ursym` being NULL if `buildArrTypeCall` is a primitive.